code/edge/src/test/java/com/adobe/marketing/mobile/EdgeExtensionTest.java

```diff
@@ -486,34 +495,121 @@ public void testhandleSetLocationHint_whenNullHint_clearsHint() {
 		assertNull(state.getLocationHint());
 	}
 
-	private void mockSharedStates(final Map<String, Object> config, final Map<String, Object> identity) {
+	@Test
+	public void testHandleSetLocationHint_whenEmptyEventData_ignoresEvent() {
+		state.setLocationHint("or2", 1800);
+		final Event requestEvent = new Event.Builder("Set Location Hint", EventType.EDGE, EventSource.UPDATE_IDENTITY)
+			.build();
+
+		edgeExtension.handleSetLocationHint(requestEvent);
+
+		assertEquals("or2", state.getLocationHint());
+	}
+
+	@Test
+	public void testHandleSetLocationHint_whenInvalidLocationHintType_doesNotCrash() {
+		state.setLocationHint("or2", 1800);
+		final Event requestEvent = new Event.Builder("Set Location Hint", EventType.EDGE, EventSource.UPDATE_IDENTITY)
+			.setEventData(
+				new HashMap<String, Object>() {
+					{
+						put("locationHint", true);
+					}
+				}
+			)
+			.build();
+
+		edgeExtension.handleSetLocationHint(requestEvent);
+
+		assertEquals("or2", state.getLocationHint());
+	}
+
+	private void mockSharedStates(
+		final SharedStateResult config,
+		final SharedStateResult identity,
+		final SharedStateResult consent
+	) {
 		when(
-			mockExtensionApi.getSharedEventState(
+			mockExtensionApi.getSharedState(
 				eq(EdgeConstants.SharedState.CONFIGURATION),
 				any(Event.class),
-				any(ExtensionErrorCallback.class)
+				any(Boolean.class),
+				any(SharedStateResolution.class)
 			)
 		)
 			.thenReturn(config);
+
 		when(
-			mockExtensionApi.getXDMSharedEventState(
+			mockExtensionApi.getXDMSharedState(
 				eq(EdgeConstants.SharedState.IDENTITY),
 				any(Event.class),
-				any(ExtensionErrorCallback.class)
+				any(Boolean.class),
+				any(SharedStateResolution.class)
 			)
 		)
 			.thenReturn(identity);
+
+		when(
+			mockExtensionApi.getXDMSharedState(
+				eq(EdgeConstants.SharedState.CONSENT),
+				any(Event.class),
+				any(Boolean.class),
+				any(SharedStateResolution.class)
+			)
+		)
+			.thenReturn(consent);
+	}
+
+	private void verifyEventQueued(final Event expectedEvent) {
+		final ArgumentCaptor<DataEntity> entityCaptor = ArgumentCaptor.forClass(DataEntity.class);
+		verify(mockQueue, times(1)).queue(entityCaptor.capture());
+		EdgeDataEntity edgeEntity = EdgeDataEntity.fromDataEntity(entityCaptor.getValue());
+		assertNotNull(edgeEntity);
+		assertEquals(expectedEvent.getName(), edgeEntity.getEvent().getName());
 	}
 
-	private void mockHubSharedState(final Map<String, Object> hub) {
+	private void verifyGetSharedStateCalls(final int configTimes, final int identityTimes, final int consentTimes) {
```

Comment:
I think it may be better if these verify calls care invoked individually inline within the unit test because the utility method masks whether a normal shared state or xdm shared state is verified, the exact event and the resolution being queried for.

Reply:
I would defer this in order to keep the test body concise and easy to read, but I can document the utility API if that helps